src/components/MAPI/Experiments.tsx

```diff
@@ -110,6 +117,8 @@ const Experiments: React.FC<IExperimentsProps> = () => {
       messageTTL.MEDIUM
     );
 
+    dispatch(clearImpersonation());
+
     reloadOrganizations();
   };
 
```

Comment:
Local state values in this component - `impersonationUser`, `impersonationGroup` - maybe it's better to take them from redux store instead of from auth provider?
```
const auth = useAuthProvider();
const [impersonationUser, setImpersonationUser] = useState<string>('');
const [impersonationGroup, setImpersonationGroup] = useState<string>('');

useEffect(() => {
    (async () => {
      const metadata = await auth.getImpersonationMetadata();

      setImpersonationUser(metadata?.user ?? '');
      setImpersonationGroup(metadata?.groups?.[0] ?? '');
    })();
}, [auth]);
```